contracts/FT-voting-simple/RIFGovernorFT.sol

```diff
@@ -5,19 +5,41 @@ import '@openzeppelin/contracts/governance/Governor.sol';
 import '@openzeppelin/contracts/governance/extensions/GovernorCountingSimple.sol';
 import '@openzeppelin/contracts/governance/extensions/GovernorVotes.sol';
 import '@openzeppelin/contracts/governance/extensions/GovernorVotesQuorumFraction.sol';
+import './ProposalTarget.sol';
 
-/// @custom:security-contact shenshin@me.com
 contract RIFGovernorFT is
     Governor,
     GovernorCountingSimple,
     GovernorVotes,
     GovernorVotesQuorumFraction
 {
+    ProposalTarget public immutable proposalTarget;
+
     constructor(IVotes _token)
         Governor('RIFGovernorFT')
         GovernorVotes(_token)
         GovernorVotesQuorumFraction(4)
-    {}
+    {
+        // deploy proposal target smart contract
+        proposalTarget = new ProposalTarget();
+    }
+
+    function execute(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) public payable override returns (uint256) {
+        uint256 proposalId = super.execute(
+            targets,
+            values,
+            calldatas,
+            descriptionHash
+        );
```

Comment:
another thing worth checking within the tests would be where
`await skipBlocks(deadline - currentBlock + 1);`
is happening, if you change this to `+ 2`, if that makes a difference - it could be linked to the `automine` config or something similar.